packages/api/models/collections.js

```diff
@@ -52,32 +57,43 @@ class Collection extends Manager {
     return super.exists({ name, version });
   }
 
+  /**
+   * Creates a collection and puts it into a collection configuration store.
+   * Uses the item's `dataType` (if specified, otherwise, it's `name`) and
+   * `version` as the key for putting the item in a config store specific to
+   * the S3 bucket and stack given by the environment variables `system_bucket`
+   * and `stackName`, respectively.
+   *
+   * @param {Object} item - the collection configuration
+   * @param {string} [item.dataType] - the collection's data type
+   * @param {string} item.name - the collection name
+   * @param {string} item.version - the collection version
+   * @see Manager#create
+   * @see CollectionConfigStore#put
+   * @returns {Promise<Object>} the created record
+   */
   async create(item) {
-    const collectionConfigStore = new CollectionConfigStore(
-      process.env.system_bucket,
-      process.env.stackName
-    );
-
-    let dataType = item.dataType;
-    if (!dataType) {
-      dataType = item.name;
-    }
-
-    await collectionConfigStore.put(dataType, item.version, item);
+    const { dataType, name, version } = item;
+    await this.collectionConfigStore.put(dataType || name, version, item);
 
     return super.create(item);
   }
 
   /**
-   * Delete a collection
+   * Deletes a collection and removes it from the corresponding collection
+   * configuration store that was used during {@link #create creation}.
    *
-   * @param {Object} params
+   * @param {Object} params - the collection configuration
```

Comment:
It's kind of semantics, but these params aren't necessarily a full collection configuration. In the primary case where this method is called from the delete endpoint, it's just an object containing the name and version.

Reply:
Makes sense, which is why I've been wondering why this method isn't simply defined with name and version as 2 distinct parameters rather than with an object parameter (especially one that has a default value of `{}`, which would seem to be useless).

Reply:
Yeah, I don't know about the default value, but I do tend to pass around objects instead of individual parameters when possible because it makes future refactoring easier. Especially in this case where it's possible we may want to pass more collection params, or an entire collection object, to the method.

Regardless, we can just change the description of `params` to be "params object" or something

Reply:
I don't agree with the general practice of preferring to define a function/method with an object parameter just for the sake of gathering up individual parameters to provide "future proofing."

Yes, there are occasions where that makes sense, particularly when there are potentially a large number of parameters, in which case this provides a "hack" for named arguments (which can aid clarity), or when we really _are_ dealing with objects.

However, taking that to the extreme, why don't we just define _every_ function/method with a single object parameter? Because that's just plain unhelpful.

As far as future refactoring a function with individual parameters, you can usually add more parameters to a function and give them sensible default values for backward compatibility. Alternatively, you can define another function with a similar name that accepts additional arguments, and the old function might call the new function with sensible defaults for the additional parameters.

Anyway, there are cleaner ways to evolve things than to simply default to defining a function with an object solely for the purpose of avoiding individual parameters.

In this particular case, since this `delete` method overrides an inherited method, we should keep the signature the same as in the superclass, including naming of the parameter, so I'm going to rename the parameter to `item` (to match the signature in the superclass) and drop the default empty object value.

Reply:
Fair enough. I really wasn't trying to suggest that we should always use objects for our parameters and in this case it's mostly a function of being called from the delete endpoint handler which receives the route params containing name/version as an object.

Regardless, +1 to renaming the paramter